src/sliding-sync.ts

```diff
@@ -615,11 +622,52 @@ export class SlidingSync extends TypedEventEmitter<SlidingSyncEvent, SlidingSync
     }
 
     /**
-     * Resend a Sliding Sync request. Used when something has changed in the request.
+     * Resend a Sliding Sync request. Used when something has changed in the request. Resolves with
+     * the transaction ID of this request on success. Rejects with the transaction ID of this request
+     * on failure.
      */
-    public resend(): void {
+    public resend(): Promise<string> {
         this.needsResend = true;
```

Comment:
Yes, which will cause the 1st request to not be sent. This is "fine" in that this class remembers what the first request contained so it will be send with the 2nd request. It's not fine as well because then the promise will not know when to resolve - it will end up rejecting after the 2nd request is resolved, which in this case is fine as that 2nd request will contain the 1st request's data, so it'll reject at the "right" time for UI purposes.

Reply:
OK - just seemed like a `if (needsResend) return <the previous promise>;` would make for more sensible behaviour (ie. it's already going to resend, so no need to schedule another one). If not, could you add a comment to this effect in the code, as this is sort of weird and surprising that the request would reject because you called resend again too quickly.

Reply:
This is a good point: if we know we haven't sent the request yet we can bundle updates more efficiently. Will fix.